prisma/schema.prisma

```diff
@@ -135,3 +136,39 @@ model ApiKey {
 
   team Team @relation(fields: [teamId], references: [id], onDelete: Cascade)
 }
+
+model Subscription {
+  id         String    @unique
+  customerId String
+  priceId    String
+  active     Boolean   @default(false)
+  startDate  DateTime
+  endDate    DateTime
+  cancelAt   DateTime?
+  createdAt  DateTime  @default(now())
+  updatedAt  DateTime  @default(now())
+}
+
+model Plan {
+  id          String   @id @default(uuid())
+  description String
+  features    String[]
+  image       String
+  name        String
+  created     DateTime
+  createdAt   DateTime @default(now())
+  updatedAt   DateTime @default(now())
+  Price       Price[]
+}
+
+model Price {
+  id            String   @id @default(uuid())
+  billingScheme String
+  currency      String
+  planId        String
+  recurring     Json
```

Comment:
The `recurring` field is a bit confusing, why can't we a `price` or `amount` field instead? and store everything else in `metadata`? `metadata` can be a JSON field, and store provider specific data. There should be a `amount` (could be 0) exists for every plans/price, so it should be a required field. What do you think?

Reply:
There are different possible plans
 - Flat price
 - Tiered price
 - Usage based
 - Package based, etc.
So not every price will have the amount
Will create a nullable amount column & save recurring in metadata